integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/automaticindexing/association/bytype/AbstractAutomaticIndexingAssociationBaseIT.java

```diff
@@ -450,6 +464,15 @@ public void directAssociationUpdate_indexedEmbedded() {
 
 			TContained oldContained = containingAssociation.get( entity1 );
 			containedAssociation.clear( oldContained );
+			containingAssociation.clear( entity1 );
```

Comment:
I think this is only necessary for single-valued associations? Could you add a `if (!isAssociationMultiValuedOnContainingSide())` (or whatever condition makes sense) to only execute this code when absolutely necessary? 

The reason: by introducing this change, you're giving Hibernate Search more "clues" that something changed, so you're helping it detect changes. I'm worried this could hide flaws in the change detection code.